cmd/root.go

```diff
@@ -34,56 +34,59 @@ var (
 	ErrIncompatibleVersion = errors.New("incompatible version")
 )
 
-var (
-	logLevel logging.Level = logging.LevelError
-	rootCmd                = &cobra.Command{
-		Use: "numerous",
-		Long: "\n                      ~~~        \n" +
-			"            ---       ~~~~~~~      \n" +
-			"     °      -------   ~~~~~~~~~~  \n" +
-			"     °°°°   ----------- ~~~~~~~~~\n" +
-			"     °°°°°°° ----------- ~~~~~~~~       _   _                                              \n" +
-			"     °°°°°°°°°°  ------- ~~~~~~~~      | \\ | |                                         \n" +
-			"     °°°°°°°°°°°°° -----  ~~~~~~~      |  \\| |_   _ _ __ ___   ___ _ __ ___  _   _ ___\n" +
-			"     °°°°°°°°°°°°° -----  ~~~~~~~      | . ` | | | | '_ ` _ \\ / _ \\ '__/ _ \\| | | / __|\n" +
-			"     °°°°°°°°°°°°° -----     ~~~~      | |\\  | |_| | | | | | |  __/ | | (_) | |_| \\__ \\\n" +
-			"     °°°°°°°°°°°°°  ----       ~~      |_| \\_|\\__,_|_| |_| |_|\\___|_|  \\___/ \\__,_|___/\n" +
-			"        °°°°°°°°°°    --\n" +
-			"          °°°°°°°°    \n" +
-			"             °°°°°   \n" +
-			"                °°     \n" +
-			"",
-		SilenceErrors: true,
-		SilenceUsage:  true,
-		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			output.NotifyFeedbackMaybe()
-
-			if !version.Check(ver.NewService(gql.NewClient())) {
-				return errorhandling.ErrorAlreadyPrinted(ErrIncompatibleVersion)
-			}
-
-			if !commandRequiresAuthentication(cmd.CommandPath()) {
-				return nil
-			}
-
-			if os.Getenv("NUMEROUS_ACCESS_TOKEN") != "" {
-				return nil
-			}
-
-			user := auth.NumerousTenantAuthenticator.GetLoggedInUserFromKeyring()
-			if user.CheckAuthenticationStatus() == auth.ErrUserNotLoggedIn {
-				output.PrintErrorLoginForCommand(cmd)
-				return ErrNotAuthorized
-			}
-
-			if err := login.RefreshAccessToken(user, http.DefaultClient, auth.NumerousTenantAuthenticator); err != nil {
-				return err
-			}
+var args = struct {
+	logLevel logging.Level
+}{
+	logLevel: logging.LevelError,
+}
 
+var cmd = &cobra.Command{
+	Use: "numerous",
+	Long: "\n                      ~~~        \n" +
```

Comment:
could we consider moving this to another file?  It takes a ridiculous amount of space

Reply:
I refactored the root command into the package `cmd/root` with files `cmd.go`, `logo.go` and `prerun.go`.